js/app/src/components/Video/Video.test.ts

```diff
@@ -0,0 +1,249 @@
+				data: null,
+				is_file: true
+			},
+			label: "Video Component",
+			root: "foo",
+			root_url: null,
+			streaming: false,
+			pending: false,
+			name: "bar",
+			source: "upload"
+		});
+		assert(queryAllByText("Video Component").length, 0);
+	});
+
+	test("static Video sets value", async () => {
+		const { getByTestId } = await render(Video, {
+			show_label: true,
+			loading_status,
+			mode: "static",
+			value: [
+				{
+					name: "https://gradio-builds.s3.amazonaws.com/demo-files/audio_sample.wav",
+					data: null,
+					is_file: true
+				}
+			],
+			root: "foo",
+			root_url: null,
+			streaming: false,
+			pending: false,
+			name: "bar",
+			source: "upload"
+		});
+		assert.isTrue(
+			getByTestId("test-player").src.endsWith(
+				"foo/file=https://gradio-builds.s3.amazonaws.com/demo-files/audio_sample.wav"
+			)
+		);
+	});
+
+	test("when autoplay is true `media.play` should be called in static mode", async () => {
+		const { getByTestId } = await render(Video, {
+			show_label: true,
+			loading_status,
+			mode: "static",
+			value: [
+				{
+					name: "https://gradio-builds.s3.amazonaws.com/demo-files/audio_sample.wav",
+					data: null,
+					is_file: true
+				}
+			],
+			root: "foo",
+			root_url: null,
+			streaming: false,
+			pending: false,
+			source: "microphone",
+			autoplay: true
+		});
+		const startButton = getByTestId<HTMLAudioElement>("test-player");
+		const fn = spyOn(startButton, "play");
+		startButton.dispatchEvent(new Event("loadeddata"));
+		assert.equal(fn.callCount, 1);
+	});
+	test("when autoplay is true `media.play` should be called in static mode", async () => {
+		const { getByTestId } = await render(Video, {
+			show_label: true,
+			loading_status,
+			mode: "dynamic",
+			value: [
+				{
+					name: "https://gradio-builds.s3.amazonaws.com/demo-files/audio_sample.wav",
+					data: null,
+					is_file: true
+				}
+			],
+			root: "foo",
+			root_url: null,
+			streaming: false,
+			pending: false,
+			source: "microphone",
+			autoplay: true
+		});
+		const startButton = getByTestId<HTMLAudioElement>("test-player");
+		const fn = spyOn(startButton, "play");
+		startButton.dispatchEvent(new Event("loadeddata"));
+		assert.equal(fn.callCount, 1);
+	});
+	test("when autoplay is true `media.play` should be called in static mode when the Video data is updated", async () => {
+		const { component, getByTestId } = await render(Video, {
+			show_label: true,
+			loading_status,
+			mode: "static",
+			value: [
+				{
+					name: "https://gradio-builds.s3.amazonaws.com/demo-files/audio_sample.wav",
+					data: null,
+					is_file: true
+				}
+			],
+			root: "foo",
+			root_url: null,
+			streaming: false,
+			pending: false,
+			source: "microphone",
+			autoplay: true
+		});
+		const startButton = getByTestId<HTMLAudioElement>("test-player");
+		const fn = spyOn(startButton, "play");
+		startButton.dispatchEvent(new Event("loadeddata"));
+		component.$set({
```

Comment:
$set is simulating the change event?

Reply:
`$set()` is the programmatic API for changing a components data.